src/autoscaler/metricscollector/collector/app_streamer.go

```diff
@@ -26,9 +26,10 @@ type appStreamer struct {
 	numRequests     map[int32]int64
 	sumReponseTimes map[int32]int64
 	ticker          clock.Ticker
+	dataChan        chan *models.AppInstanceMetric
 }
 
-func NewAppStreamer(logger lager.Logger, appId string, interval time.Duration, cfc cf.CfClient, noaaConsumer noaa.NoaaConsumer, database db.InstanceMetricsDB, sclock clock.Clock) AppCollector {
+func NewAppStreamer(logger lager.Logger, appId string, interval time.Duration, cfc cf.CfClient, noaaConsumer noaa.NoaaConsumer, database db.InstanceMetricsDB, sclock clock.Clock, dataChan chan *models.AppInstanceMetric) AppCollector {
```

Comment:
Currently the metric data will put into dataChan, so the parameter "database" is useless.